api/v1alpha1/clusterextension_types.go

```diff
@@ -69,100 +68,93 @@ type ClusterExtensionSpec struct {
+	// appropriate bundle of content to install will be fetched from
+	// ClusterCatalog resources existing on the cluster.
+	// When using the Catalog sourceType, the catalog field must also be set.
 	//
 	// +unionDiscriminator
 	// +kubebuilder:validation:Enum:="Catalog"
+	// +kubebuilder:validation:Required
 	SourceType string `json:"sourceType"`
 
-	// catalog is used to configure how information is sourced from a catalog. This field must be defined when sourceType is set to "Catalog",
-	// and must be the only field defined for this sourceType.
+	// catalog is used to configure how information is sourced from a catalog.
+	// This field is required when sourceType is "Catalog", and forbidden otherwise.
 	//
 	// +optional.
 	Catalog *CatalogSource `json:"catalog,omitempty"`
 }
 
 // ClusterExtensionInstallConfig is a union which selects the clusterExtension installation config.
 // ClusterExtensionInstallConfig requires the namespace and serviceAccount which should be used for the installation of packages.
+//
 // +union
 type ClusterExtensionInstallConfig struct {
-	// namespace is a reference to the Namespace in which the bundle of
-	// content for the package referenced in the packageName field will be applied.
+	// namespace designates the kubernetes Namespace where bundle content
+	// for the 'packageName' package field will be applied and the necessary
+	// service account can be found.
 	// The bundle may contain cluster-scoped resources or resources that are
 	// applied to other Namespaces. This Namespace is expected to exist.
 	//
 	// namespace is required, immutable, and follows the DNS label standard
-	// as defined in [RFC 1123]. This means that valid values:
-	//   - Contain no more than 63 characters
-	//   - Contain only lowercase alphanumeric characters or '-'
-	//   - Start with an alphanumeric character
-	//   - End with an alphanumeric character
-	//
-	// Some examples of valid values are:
-	//   - some-namespace
-	//   - 123-namespace
-	//   - 1-namespace-2
-	//   - somenamespace
-	//
-	// Some examples of invalid values are:
-	//   - -some-namespace
-	//   - some-namespace-
-	//   - thisisareallylongnamespacenamethatisgreaterthanthemaximumlength
-	//   - some.namespace
+	// as defined in [RFC 1123]. It must contain only lowercase alphanumeric characters or hyphens (-),
+	// start and end with an alphanumeric character, and be no longer than 63 characters
 	//
 	// [RFC 1123]: https://tools.ietf.org/html/rfc1123
 	//
-	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
-	//+kubebuilder:validation:MaxLength:=63
-	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="namespace is immutable"
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="namespace is immutable"
+	// +kubebuilder:validation:XValidation:rule="self.matches(\"^[a-z0-9]([-a-z0-9]*[a-z0-9])?$\")",message="namespace must be a valid DNS1123 label"
```

Comment:
I know we describe what a RFC 1123 DNS label is in the description of this field, but a user that inputs an incorrect value won't know how to fix "namespace must be a valid DNS1123 label" without some additional searching. Should we add some additional information in the message such that a user can more easily understand what the validation criteria is from the admission error message they receive?

Maybe something like:
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.matches(\"^[a-z0-9]([-a-z0-9]*[a-z0-9])?$\")",message="namespace must be a valid DNS1123 label. Valid DNS1123 labels contain only lowercase alphanumeric characters or hyphens, starting and ending with an alphanumeric character"
```

Reply:
How important is this, given that the godoc has the details?  I'm curious because my first impulse when I get a message like this would be to web search 'RFC 1123 label', which would pretty quickly get to the answer.

Reply:
I wouldn't block the PR on this, but my thinking is that while we are making the change we might as well make sure we are providing as helpful of an error message to the end user as possible. While doing a web search may be many of our first impulse, I would like to think many of us would prefer error messages that clearly signal the general requirements of the field such that it saves us needing to perform that additional web search step.

This is purely a nice-to-have from the end user perspective and I would rank it high on the importance scale, but not high enough to block this since updating the message to be "nicer" would be a non-breaking change.